code/game/machinery/teleporter.dm

```diff
@@ -225,3 +219,125 @@
 		return ..()
 	icon_state = base_icon_state
 	return ..()
+
+/obj/machinery/teleport/syndicate_gate
+	icon = 'icons/obj/machines/teleporter_multitile.dmi'
+	icon_state = "teleporter_off"
+	pixel_x = -32
+	///Tracks wether the portal is active or not, used to toggle the sprite
+	var/activated = FALSE
+	///When a painting targets us as their signal, we save them as our return target
+	var/atom/return_target
+	///Internal timer to prevent audio spam.
+	var/next_beep = 0
+
+/obj/machinery/teleport/syndicate_gate/Destroy()
+	GLOB.active_syndicate_gates -= src
+	return_target = null
+	return ..()
+
+/obj/machinery/teleport/syndicate_gate/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_PAINTING_SET_TARGET, PROC_REF(on_target_set))
+	RegisterSignals(src, list(COMSIG_QDELETING, COMSIG_MACHINERY_BROKEN, COMSIG_PAINTING_CUT_CONNECTIONS), PROC_REF(remove_connections))
+
+/obj/machinery/teleport/syndicate_gate/Bumped(mob/living/user)
+	if(!ishuman(user))
+		return //Otherwise the sparks keep making the sound spam
+	if(!return_target)
+		if(next_beep <= world.time)
+			next_beep = world.time + (2 SECONDS)
+			playsound(src, 'sound/machines/scanbuzz.ogg', 100, FALSE)
+		return
+
+	if(user.mind.has_antag_datum(/datum/antagonist/satellite_agent))
+		var/confirmation = tgui_alert(user, "Are you sure you wish to leave the satellite, this should only be a last resort to help a field agent", "WARNING", list("Teleport?", "cancel"))
```

Comment:
they should not be able to leave at all.
ghost roles should have zero impact on that station

Reply:
Xander Fumbled the description of the pr.

It's not a selectable ghost role, it's an antag you summon via an uplink item that's picked from the ghost pool, same as nightmare, dragon and so on.

Reply:
So yeah I explained poorly.
This isn't a ghost role, you are effectively paying 10 TC for a reinforcement that mainly babysits the satellite.
They are specifically allowed to leave but only when the circumstances require it since leaving haphazardly means that you leave the base undefended.